packages/components/src/DateTimePickers/DateRange/Manager/Manager.component.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { DateRangeContext } from '../Context';
+import {
+	extractRangeParts,
+	extractRangePartsFromTextInput,
+	extractRangePartsFromDate,
+} from '../date-range-extraction';
+
+function ContextualManager(props) {
+	function getOptions() {
+		return {
+			dateFormat: props.dateFormat,
+		};
+	}
+	const initialState = extractRangeParts(props.startDate, props.endDate, getOptions());
+	const [state, setState] = useState(initialState);
+
+	useEffect(() => {
+		if (props.startDate !== state.startDate.value || props.endDate !== state.endDate.value) {
+			const parts = extractRangeParts(props.startDate, props.endDate, getOptions());
+			setState(parts);
+		}
+	}, [props.startDate, props.endDate]);
+
+	function onDatesChange(event, nextState) {
+		if (props.onChange) {
+			const payload = {
+				startDate: nextState.startDate.value,
+				endDate: nextState.endDate.value,
+				errors: nextState.errors,
+				errorMessage: nextState.errorMessage,
+				field: state.focusedInput,
+				origin: 'RANGE_PICKER',
```

Comment:
What is this for ?

Reply:
It will be used later in custom hook of InputDateRangePicker:
https://github.com/Talend/ui/blob/master/packages/components/src/DateTimePickers/hooks/useInputPickerHandlers.js#L25
to distinguish with `origin=PICKER`, 
when use single date picker, after user pick a date, it will close picker and focus on the input, for range picker, after picker a date, I want to move focus to endDate.

Reply:
In this case, please set a difference between start and end, so you can apply the focus only on START_PICKER origin. Don't set a common `RANGE_PICKER` value for both, where you'll struggle to check which one changed

Reply:
changed to use `START_PICKER` and `END_PICKER`